### ISSUE TITLE
Compatible with react-hot-loader

### ISSUE THREAD

**Issue by yesmeck** [ant-design/ant-design#4655]
Related issues: #4651 #3550 #4404 #4075 

Components need works:

- [x] Breadcrumb
- [x] rc-select
- [x] rc-tree
- [x] rc-table

Solution:

```javascript
class BreadcrumbItem extends React.Component {
  static __ANT_BREADCRUMB_ITEM = true;
}


// use 
if (element.type.__ANT_BREADCRUMB_ITEM) {
    ...
}
```

**Comment by afc163** [comment]
similar issue for HOC component: https://github.com/callemall/material-ui/issues/6041

**Comment by yesmeck** [comment]
HOC，还有 Composition 的问题，除了用  type 做比较外，我们还大量使用了 cloneElement，有些地方其实用  context  会更好点。

https://github.com/react-component/menu/pull/72

**Comment by yesmeck** [comment]
Released in 2.7.0.
